packages/material-ui-lab/src/Breadcrumbs/Breadcrumbs.js

```diff
@@ -0,0 +1,141 @@
+import React, { Children } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import classNames from 'classnames';
+import BreadcrumbCollapsed from '@material-ui/lab/BreadcrumbCollapsed';
+import BreadcrumbSeparator from '@material-ui/lab/BreadcrumbSeparator';
+
+const styles = {
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+  },
+};
+
+class Breadcrumbs extends React.Component {
+  state = {
+    expanded: false,
+  };
+
+  handleClickExpand = event => {
+    event.preventDefault();
+    this.setState({ expanded: true });
+  };
+
+  insertSeparators(items) {
+    const { separator, separatorText } = this.props;
+
+    return items.reduce(
+      (acc, cur, idx, src) =>
+        idx < src.length - 1
+          ? acc.concat(
+              cur,
+              <BreadcrumbSeparator
+                key={`separator-${idx}`}
```

Comment:
@eps1lon Any suggestions what to replace index with here, or is it safe to add an exception? I can't see the order changing...

Reply:
I think it's quite ok to use the index here if we can't make any assumptions about the semantics. For something like breadcrumbs it's fairly unlikely that they change the order. You might add or remove items but you only do this to the end so you don't end up in unnecessary rerenders. 

It's a very light weight component I would say so you shouldn't worry to much about perf. Sure if you use this to separate multiple very long tables and splice those up frequently then you will encounter deoptimizations. I don't think that's a common use case though.

Reply:
Okay great, that echos what I wrote originally, but edited out so as not to “lead the witness”.

Please could you give the rest of the component code another review? (The examples are unchanged).

Once that’s stable I’ll add tests, but might need help with types.